Libraries/Opc.Ua.Server/Configuration/ConfigurationNodeManager.cs

```diff
@@ -553,8 +566,24 @@ private ServiceResult CreateSigningRequest(
                 throw new ArgumentNullException(nameof(subjectName));
             }
 
-            // TODO: implement regeneratePrivateKey
-            // TODO: use nonce for generating the private key
+
+            certificateGroup.TemporaryApplicationCertificate?.Dispose();
+            certificateGroup.TemporaryApplicationCertificate = null;
+
+            if (regeneratePrivateKey)
+            {
+                X509Certificate2 newCert = CertificateFactory.CreateCertificate(
+                m_configuration.ApplicationUri,
+                m_configuration.ApplicationName,
+                certificateGroup.ApplicationCertificate.SubjectName,
+                X509Utils.GetDomainsFromCertificate(certificateGroup.ApplicationCertificate.Certificate))
+                .CreateForRSA();
```

Comment:
We can´t garuantee the private key has the same size as the private key of the existing ApplicationCertificate. How do we extract this information from the existing Certificate? 
The lifetime will also be the default, should we read this from the existing ApplicationCertificate